src/app/user/user.service.ts

```diff
@@ -203,4 +211,13 @@ export class UserService {
   resetUser(): void {
     this.userData = {} as User;
   }
+
+  private catchRequestError() {
```

Comment:
If you made this a property whose value is an arrow function, then you wouldn't have to return a new function every time:

`private catchRequestError = (response: Response) => {`

Then you can simply do the following for all your `catch` statements:
`.catch(this.catchRequestError)`